data/json/items/melee/bludgeons.json

```
@@ -935,5 +934,39 @@
     "cutting": 23,
     "price": "160 USD",
     "qualities": [ [ "HAMMER", 2 ] ]
+  },
+  {
+    "id": "qstaff_retracted",
+    "type": "GENERIC",
+    "category": "weapons",
+    "looks_like": "baton",
+    "weapon_category": [ "BATONS" ],
+    "name": { "str": "extendable quarterstaff (retracted)", "str_pl": "extendable quarterstaves (retracted)" },
+    "description": "A custom-made telescoping steel quarterstaff, capable of shrinking down to the size of a baton for easy storage. Its hollow nature makes it more lightweight but less sturdy than a more traditional staff. It is currently retracted.",
+    "weight": "2000 g",
+    "volume": "750 ml",
+    "price_postapoc": "10 USD",
+    "color": "dark_gray",
+    "symbol": "/",
+    "material": [ "steel" ],
+    "techniques": [ "WBLOCK_1", "RAPID" ],
+    "flags": [ "BELT_CLIP" ],
+    "use_action": { "menu_text": "Expand", "type": "transform", "target": "qstaff_extended", "msg": "You snap open your staff." },
+    "bashing": 14
+  },
+  {
+    "id": "qstaff_extended",
+    "type": "GENERIC",
+    "copy-from": "qstaff_retracted",
+    "looks_like": "i_staff",
+    "weapon_category": [ "QUARTERSTAVES" ],
+    "name": { "str": "extendable quarterstaff (extended)", "str_pl": "extendable quarterstaves (extended)" },
+    "description": "A custom-made telescoping steel quarterstaff, capable of shrinking down to the size of a baton for easy storage. Its hollow nature makes it more lightweight but less sturdy than a more traditional staff. It is currently extended.",
+    "volume": "3 L",
+    "flags": [ "REACH_ATTACK" ],
```

Probably shouldn't add reach attack unless one plans to add them to other staffs in this PR. Speaking of things that are weird about other staves that maybe could be tweaked in a later PR:
```suggestion
    "flags": [ "SHEATH_SPEAR", "ALWAYS_TWOHAND" ],
```
The former depends on if we want it to change up whether it belongs on a belt clip or a spear strap depending on its state, or if we just want it to only be stowable in retracted form. The latter of course is just since for now that's what other staves have.